python/kfserving/kfserving/storage.py

```diff
@@ -173,7 +172,9 @@ def _download_blob(uri, out_dir: str):  # pylint: disable=too-many-locals
                     os.makedirs(dir_path)
 
             logging.info("Downloading: %s to %s", blob.name, dest_path)
-            block_blob_service.get_blob_to_path(container_name, blob.name, dest_path)
+            downloader = blob_service_client.get_container_client(container_name).download_blob(blob.name)
```

Comment:
`get_container_client` gets called twice here.

Reply:
Yup, will take care of this. Let's continue the discussion on how to allow users to setup the AZURE_STORAGE_CONNECTION_STRING, as this will slightly broaden the scope of this PR.